src/crate/client/sqlalchemy/types.py

```diff
@@ -132,7 +132,7 @@ def __eq__(self, other):
 
 
 class _Craty(sqltypes.UserDefinedType):
-    cache_ok = True
+    cache_ok = False
```

Comment:
The effective change is this one. The other code is for testing only.

Reply:
It may be advisable to look at SQLAlchemy's implementation for PostgreSQL's [JSONB](http://docs.sqlalchemy.org/en/latest/dialects/postgresql.html#sqlalchemy.dialects.postgresql.JSONB) type, and its [`contains()`](http://docs.sqlalchemy.org/en/latest/dialects/postgresql.html#sqlalchemy.dialects.postgresql.JSONB.Comparator.contains) method, or at the corresponding implementation for MSSQL's [JSON](https://docs.sqlalchemy.org/en/latest/dialects/mssql.html#sqlalchemy.dialects.mssql.JSON) type.

-- https://stackoverflow.com/a/39467149

Reply:
I played a bit with the `_Craty` implementation for `OBJECT` types within the `crate.client.sqlalchemy.types` module, trying to make this statement work, which I believe would be correct in that sense, and where I would hope the SQLAlchemy machinery would handle it well, without getting the statement caching wrong.
```python
selectable = sa.select(Character).where(Character.data['x'].as_integer() == 1)
```

However, I got different exceptions, so I think it will need further investigations to make this work. In that manner, I suggest to postpone it to a later iteration, and release the bugfix without further ado.

Reply:
GH-560 elaborates on what has been suggested above.